### ISSUE TITLE
Support AnyCPU on .NET Framework applications

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/libuv-package#21]
Currently this package does not support AnyCPU which means .NET Framework apps _must_ use PlatformTarget=x64 or x86.

It is fairly easy to workaround this. We did this in the Microsoft.Data.Sqlite package by (1) adding a little managed code that locates sqlite3.dll and loads it via `LoadLibraryEx` and (2) adding an MSBuild target in the package to copy files to the output folder when PlatformTarget=AnyCPU.

Example:
https://github.com/natemcmaster/libsqlite3-package/blob/ba16e2ede7768cf80cfe7a6023109e2c5a59e30e/files/SQLite.props
https://github.com/aspnet/Microsoft.Data.Sqlite/blob/rel/1.1.1/src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs

**Issues**
https://github.com/aspnet/KestrelHttpServer/issues/1625
https://github.com/aspnet/KestrelHttpServer/issues/1652
https://github.com/aspnet/KestrelHttpServer/issues/696

cc @davidfowl @moozzyk @muratg

**Comment by muratg** [comment]
I wonder if this is the cause of the recent scaffolding issue on a 32 bit machine.  cc @mlorbetske

**Comment by davidfowl** [comment]
I was hoping this would go away when we changed to sockets by default. Maybe it's fine to do this to make .NET Framework applications work better.

TL;DR I would *only* do this for .NET Framework running on windows. I wouldn't change the general native library loading for anything else.

@natemcmaster does that props file only apply to .NET Framework as well?

**Comment by natemcmaster** [comment]
> does that props file only apply to .NET Framework as well?

Correct. The MSBuild props is _only_ for .NET Framework. It also means a user must have a PackageReference to libuv as MSBuild assets do not flow across project-to-project references even though native assets do. But this generally hasn't been a problem for SQLite.

**Comment by moozzyk** [comment]
Does it mean that Kestrel needs to handle loading of libuv directly instead of using DllImport?

**Comment by natemcmaster** [comment]
We can still use `[DllImport("libuv")]`. But before we invoke any of these extern methods we would need to manually find and load libuv.dll using [`LoadLibraryEx`](https://github.com/aspnet/Microsoft.Data.Sqlite/blob/15cd7845b8c42cfb0347134e640d2733e9a65de6/src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs#L48). This piece of managed code could live in the Libuv package itself, or it could be in Kestrel.

Something like this:
```c#
namespace Libuv.Interop
{
    public class LibuvLoader
    {
         public static bool TryLoadNativeLibrary()
         {
            // ...
         }
    }
}

```

**Comment by davidfowl** [comment]
Right. Only on .net Framework on windows.id rather not have to figure out native library loading for all platforms when the CLR already does it.

**Comment by muratg** [comment]
@moozzyk Low priority for now.
